src/Features/CSharp/Portable/Copilot/CSharpCopilotCodeFixProvider.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Copilot;
+using Microsoft.CodeAnalysis.CSharp.LanguageService;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.CopilotSuggestions), Shared]
+internal sealed partial class CSharpCopilotCodeFixProvider : CopilotCodeFixProvider
+{
+    private const string FixPropertyName = "Fix";
+    private const string PromptTitlePropertyName = "PromptTitle";
+
+    private static SyntaxAnnotation WarningAnnotation { get; }
+        = CodeActions.WarningAnnotation.Create(
+            CSharpFeaturesResources.Warning_colon_AI_suggestions_might_be_inaccurate);
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpCopilotCodeFixProvider()
+    {
+    }
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var document = context.Document;
+        var cancellationToken = context.CancellationToken;
+
+        var copilotService = document.Project.Solution.Services.GetService<ICopilotCodeAnalysisService>();
+        if (copilotService is null || !copilotService.IsCodeAnalysisOptionEnabled(document))
+            return;
+
+        var isAvailable = await copilotService.IsAvailableAsync(document, cancellationToken).ConfigureAwait(false);
+        if (!isAvailable)
+            return;
+
+        var promptTitles = await copilotService.GetAvailablePromptTitlesAsync(document, cancellationToken).ConfigureAwait(false);
+        if (promptTitles.IsEmpty)
+            return;
+
+        var hasMultiplePrompts = promptTitles.Length > 1;
+
+        // Find the containing method, if any, and also update the fix span to the entire method.
+        // TODO: count location in doc-comment as part of the method.
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var containingMethod = CSharpSyntaxFacts.Instance.GetContainingMethodDeclaration(root, context.Span.Start, useFullSpan: false);
```

Comment:
same issue with methods vs other constructs (like properties).  i don't understand this filtering and it isn't doc'ed .